pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -266,69 +280,41 @@ func (d *DualWriterMode1) Update(ctx context.Context, name string, objInfo rest.
 	ctx = klog.NewContext(ctx, log)
 
 	startLegacy := time.Now()
-	res, async, err := d.Legacy.Update(ctx, name, objInfo, createValidation, updateValidation, forceAllowCreate, options)
+	objLegacy, async, err := d.Legacy.Update(ctx, name, objInfo, createValidation, updateValidation, forceAllowCreate, options)
+	d.recordLegacyDuration(err != nil, mode1Str, d.resource, method, startLegacy)
 	if err != nil {
 		log.Error(err, "unable to update in legacy storage")
-		d.recordLegacyDuration(true, mode1Str, d.resource, method, startLegacy)
-		return res, async, err
+		return objLegacy, async, err
 	}
-	d.recordLegacyDuration(false, mode1Str, d.resource, method, startLegacy)
 
 	//nolint:errcheck
-	go d.updateOnUnifiedStorage(ctx, res, name, objInfo, createValidation, updateValidation, forceAllowCreate, options)
+	go d.updateOnUnifiedStorage(ctx, objLegacy, name, objInfo, createValidation, updateValidation, forceAllowCreate, options)
 
-	return res, async, err
+	return objLegacy, async, err
 }
 
-func (d *DualWriterMode1) updateOnUnifiedStorage(ctx context.Context, res runtime.Object, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) error {
+func (d *DualWriterMode1) updateOnUnifiedStorage(ctx context.Context, objLegacy runtime.Object, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) error {
 	var method = "update"
 	log := d.Log.WithValues("name", name, "method", method, "name", name)
 
 	// Ignores cancellation signals from parent context. Will automatically be canceled after 10 seconds.
 	ctx, cancel := context.WithTimeoutCause(context.WithoutCancel(ctx), time.Second*10, errors.New("storage update timeout"))
 
-	resCopy := res.DeepCopyObject()
```

Comment:
Nice !

Reply:
I'd rather keep it for metrics, does that make sense?